source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2386,6 +2386,48 @@ lldb::addr_t SwiftLanguageRuntime::FixupAddress(lldb::addr_t addr,
   return addr;
 }
 
+const swift::reflection::TypeInfo *
+SwiftLanguageRuntime::GetTypeInfo(CompilerType type) {
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  ConstString mangled_name(type.GetMangledTypeName());
+  StringRef mangled_no_prefix =
+      swift::Demangle::dropSwiftManglingPrefix(mangled_name.GetStringRef());
+  swift::Demangle::Demangler Dem;
+  auto demangled = Dem.demangleType(mangled_no_prefix);
+  auto *type_ref = swift::Demangle::decodeMangledType(
+      reflection_ctx->getBuilder(), demangled);
+  if (!type_ref)
+    return nullptr;
+  return reflection_ctx->getBuilder().getTypeConverter().getTypeInfo(type_ref);
+}
+
+bool SwiftLanguageRuntime::IsIndirectResilientType(CompilerType type) {
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  swift::NominalType *nominal_type =
+      swift_can_type->getAs<swift::NominalType>();
+  if (!nominal_type)
+    return false;
+
+  swift::NominalTypeDecl *nominal_type_decl = nominal_type->getDecl();
+  if (!nominal_type_decl)
+    return false;
+
+  if (!nominal_type_decl->isResilient())
+    return false;
+
+  auto *type_info = GetTypeInfo(type);
+  if (!type_info)
+    return false;
+  return !type_info->isBitwiseTakable() || type_info->getSize() > 24;
```

Comment:
As in the other PR, the 24 bytes is actually only valid on 64-bit, you should just use `word size * 3` instead.